src/main/java/se/sundsvall/casedata/service/ConversationService.java

```diff
@@ -0,0 +1,131 @@
+			.orElseThrow(() -> Problem.valueOf(INTERNAL_SERVER_ERROR, "Failed to create conversation in Message Exchange"));
+
+		final var conversationEntity = toConversationEntity(conversation, municipalityId, namespace, errandId, messageExchangeId);
+		return conversationRepository.save(conversationEntity).getId();
+	}
+
+	public Conversation getConversation(final String municipalityId, final String namespace, final Long errandId, final String conversationId) {
+
+		final var entity = getConversationEntity(municipalityId, namespace, errandId, conversationId);
+
+		final var response = messageExchangeClient.getConversation(municipalityId, namespace, entity.getMessageExchangeId());
+
+		if (!response.getStatusCode().is2xxSuccessful()) {
+			throw Problem.valueOf(INTERNAL_SERVER_ERROR, "Failed to retrieve conversation from Message Exchange");
+		}
+
+		if (response.getBody() == null) {
+			throw Problem.valueOf(NOT_FOUND, "Conversation not found in Message Exchange");
+		}
+
+		return toConversation(entity, response.getBody());
+	}
+
+	public List<Conversation> getConversations(final String municipalityId, final String namespace, final Long errandId) {
+
+		final var conversations = conversationRepository.findByMunicipalityIdAndNamespaceAndErrandId(municipalityId, namespace, errandId.toString());
+
+		if (conversations.isEmpty()) {
+			throw Problem.valueOf(NOT_FOUND, "No conversations found for the given parameters");
+		}
+
+		return conversations.stream()
+			.map(conversationEntity -> {
+
+				final var response = messageExchangeClient.getConversation(municipalityId, namespace, conversationEntity.getMessageExchangeId());
+
+				if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
+					throw Problem.valueOf(INTERNAL_SERVER_ERROR, "Failed to retrieve conversation from Message Exchange");
+				}
+				return toConversation(conversationEntity, response.getBody());
+			})
+			.toList();
+
+	}
+
+	public Conversation updateConversation(final String municipalityId, final String namespace, final Long errandId, final String conversationId, final Conversation request) {
+
+		final var entity = getConversationEntity(municipalityId, namespace, errandId, conversationId);
+
+		final var response = messageExchangeClient.updateConversation(municipalityId, namespace, entity.getMessageExchangeId(), toMessageExchangeConversation(request));
+
+		if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
+			throw Problem.valueOf(INTERNAL_SERVER_ERROR, "Failed to update conversation in Message Exchange");
+		}
+
+		updateConversationEntity(entity, request);
+		conversationRepository.save(entity);
+		return toConversation(entity, response.getBody());
+	}
+
+	public void createMessage(final String municipalityId, final String namespace, final long errandId, final String conversationId, final Message messageRequest, final List<MultipartFile> attachments) {
```

Comment:
If i remember correctly the attachments should also be saved on errand. Similar to email and webmessage handling.

Reply:
I'll handle that in a different task. (UF-16383)